balancer-labs/frontend-v2
# Feat: front-end allowlist for weighted pools with rate providers

# Description

For weighted pools, we check if `priceRateProviders` field from subgraph, which returns rate providers for pool tokens, includes any rate providers which we haven't allow-listed in the front-end. If that's the case, we show warning.

## Type of change


## How should this be tested?

Go to pool ethereum/pool/0x25accb7943fd73dda5e23ba6329085a3c24bfb6a000200000000000000000387
It shouldn't show any warnings. Go to other weighted pool with non-allow-listed rate providers - there should be warning on pool page.


## Visual context

N.A

## Checklist:
